recipe/meta.yaml

```diff
@@ -30,7 +30,8 @@ requirements:
   run:
     - python >={{ python_min }},<3.13
     - pip
-    - numpy >=1.21.0,<2.0.0
+    - numpy <3.0,>=2.0  # py>=3.12
+    - numpy >=1.21.0,<2.0.0  # py<3.12
```

Comment:
@conda-forge/core Are selectors supported for `noarch` packages? I remember this was not the case in the past but at least the linter does not complain.

Reply:
They shouldn't be supported as only a single package is going to be created and that will contain only one of your two options. At prefix we've just started to work on optinal / conditional dependencies, so hopefully relatively soon we can fix this properly for all of conda.

Reply:
Yes, I noticed that I was missing the `[]`. After adding those I got the expected error message. So at the moment the only solution is to remove `noarch` and then use the selectors, correct?

Reply:
I think so, yes :/

Reply:
No, that's not the only solution. There's a section on noarch packages on conda-forge docs. (On my phone, can't link)

Reply:
@isuruf are you referring to https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies ?

Reply:
I found the documentation page https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python but the only thing it mentions is to restrict the python version directly. I do not understand how to implement coupled dependencies like:
```
[tool.poetry.dependencies]
python = ">=3.9,<3.14"
numpy = [
    { version = "^2.0", python = ">=3.12" },
    { version = "^1.21", python = "<3.12" },
]
```

Reply:
It could be done by creating two outputs via variants. You could define a variant `python_larger_312: [true, false]` and then add

```
- numpy 2.* # [python_larger_312 == true]
- python >=3.12 # [python_larger_312 == true]
- numpy 1.21 # [python_larger_312 == false]
- ...
```

Reply:
You can create two packages with a `recipe/conda_build_config.yaml` like
```
with_python_gt_312:
 - True
 - False
```
and then use those selectors

Reply:
I did the change - the tests seem to work fine, but the linter complains. Can I merge it anyway?